### DISCUSSION TITLE
Consequences of disabling fstrim?

### DISCUSSION THREAD

**Discussion by ondrejmo** [minio/minio#20254] [Q&A] [answered]
Hi, I have a question about the `fstrim` [being disabled](https://min.io/docs/minio/linux/operations/checklists/software.html#minio-pre-requisites) on the drives used by MinIO. I am aware of [the consequences](https://github.com/minio/minio/discussions/17311) of keeping `fstrim` enabled, but generally speaking trimming should be done on SSDs so that the memory cells are used evenly and their performance / longevity does not degrade. 

Does MinIO somehow replace the fstrim's behavior and if not does it have any long-term affects on longevity and performance of the drives? The drives in question are Samsung PM9A3 15,36 TB formatted as XFS.

**Comment by klauspost** [comment]
> Does MinIO somehow replace the fstrim's behavior...

No. There are no special considerations by disabling it. The behavior will be the same as any other application, and the consequence of disabling it will depend on your hardware.

**Comment by harshavardhana** [comment, marked as answer]
From our discussions with NVMe vendors, apparently they don't want you to run fstrim at software level anymore.

The fstrim like functionality is apparently internal to their drives architecture and does not require any host level settings.

Always check with your drive vendor.
